test-network-function/generic/suite.go

```diff
@@ -510,6 +586,21 @@ func testBootParams(context *interactive.Context, podName, podNamespace string,
 	})
 }
 
+func testSysctlConfigs(context *interactive.Context, podName, podNamespace string) {
+	ginkgo.It(fmt.Sprintf("Testing sysctl config files for the pod's node %s/%s", podNamespace, podName), func() {
+		nodeName := getPodNodeName(context, podName, podNamespace)
+		combinedSysctlSettings := getSysctlConfigArgs(context, nodeName)
+		mcName := getMcName(context, nodeName)
+		mcKernelArgumentsMap := getMcKernelArguments(context, mcName)
+
+		for key, sysctlConfigVal := range combinedSysctlSettings {
```

Comment:
I don't know enough about how the two different sets of values interact to know if this is correct but currently this is not asserting in all cases.

Currently if a value is:
1. present in both maps and equal -> assertion passes
2. present in both maps but unequal -> assertion fails
3. present only in `sysctlConfigVal` -> triggers no assertion, therefore passes
4. present only in `mcKernelArgumentsMap` -> triggers no assertion, therefore passes

@ryandgoulding do you know if the current behaviour is correct?

Reply:
You understand it correctly.
This is the way it supposed to behave.